pkg/runtime/processor/processor_test.go

```diff
@@ -654,3 +654,51 @@ func TestReporter(t *testing.T) {
 			}
 		})
 }
+
+func TestProcessorWaitGroupError(t *testing.T) {
+	ctx := context.Background()
+	proc, _ := newTestProc()
+	// spin up the processor
+	go func() {
+		proc.Process(ctx)
+	}()
```

Comment:
Always ensure errors are checked and all go rountines are returned at the end of the test. Does this ever return?

Reply:
it does not